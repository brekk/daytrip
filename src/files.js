import fg from 'fast-glob'
import { Future } from 'fluture'

export const glob = input =>
  new Future((bad, good) => {
    fg(input).catch(bad).then(good)
    /* istanbul ignore next */
    return function cancel() {}
  })

