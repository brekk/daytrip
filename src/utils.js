import { fork as rawFork } from 'fluture'
import { curry } from 'ramda'

export const fork = curry((bad, good, future) =>
  rawFork(bad)(good)(future)
)
export const pretack = curry((raw, x) => [raw, x])
