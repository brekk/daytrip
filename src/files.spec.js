import { glob } from './files'
import { fork } from './utils'

test('glob', done => {
  const futureGlob = glob(`${__dirname}/*.js`)
  fork(
    done,
    output => {
      expect(output).toMatchSnapshot()
      done()
    },
    futureGlob
  )
})
