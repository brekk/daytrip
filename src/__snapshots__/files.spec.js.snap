// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`glob 1`] = `
Array [
  "/Users/brekk/work/daytrip/src/daytrip.js",
  "/Users/brekk/work/daytrip/src/daytrip.spec.js",
  "/Users/brekk/work/daytrip/src/dependencies.js",
  "/Users/brekk/work/daytrip/src/files.js",
  "/Users/brekk/work/daytrip/src/files.spec.js",
  "/Users/brekk/work/daytrip/src/index.js",
  "/Users/brekk/work/daytrip/src/string.js",
  "/Users/brekk/work/daytrip/src/string.spec.js",
  "/Users/brekk/work/daytrip/src/utils.js",
  "/Users/brekk/work/daytrip/src/utils.spec.js",
]
`;
